module/VuFind/src/VuFind/View/Helper/Root/RecordTabs.php

```diff
@@ -0,0 +1,107 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\View\Helper\Root;
+
+use VuFind\RecordTab\TabManager;
+
+use function in_array;
+
+/**
+ * Record tab view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class RecordTabs extends \Laminas\View\Helper\AbstractHelper
+{
+    /**
+     * Constructor
+     *
+     * @param array      $config     Config
+     * @param TabManager $tabManager Tab Manager
+     */
+    public function __construct(protected array $config, protected TabManager $tabManager)
+    {
+    }
+
+    /**
+     * Render record tabs.
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $driver    Record driver
+     * @param array                             $tabs      Tabs
+     * @param string                            $activeTab Active tab
+     *
+     * @return array
+     */
+    public function __invoke(
+        \VuFind\RecordDriver\AbstractBase $driver,
+        array $tabs,
+        string $activeTab
+    ): array {
+        $ajaxTabUrl = $this->getView()->recordLinker()->getTabUrl($driver, 'AjaxTab');
+        $loadInitialTabWithAjax = (bool)($this->config['Site']['loadInitialTabWithAjax'] ?? false);
+        $backgroundTabs = $this->tabManager->getBackgroundTabNames($driver);
+        $tabArray = [];
+        foreach ($tabs as $tab => $obj) {
+            $tabItem = [];
+            $tabName = preg_replace("/\W/", '-', strtolower($tab));
+            $tabItem['description'] = $obj->getDescription();
+            $tabItem['visible'] = $obj->isVisible();
+            $tabItem['buttonAttributes'] = [
+                'class' => 'record-tab-button',
+            ];
+
+            $isActiveTab = $activeTab === $tab;
+            $tabContent = '';
+            if (($isActiveTab && !$loadInitialTabWithAjax) || !$obj->supportsAjax()) {
+                $tabContent = $this->getView()->record($driver)->getTab($obj);
+            }
+
+            $tabItem['content'] = $tabContent;
+            $tabItem['paneAttributes'] = [
+                'class' => 'record-tab-pane',
+                'data-ajax-url' => $ajaxTabUrl,
+            ];
+            if (!empty($tabContent) || !$obj->supportsAjax()) {
```

Comment:
I find this a little confusing -- especially the different if statements on line 86 and 95 that I _think_ are trying to capture the same condition. Does this minor rewrite clarify/simplify things without breaking them, or am I missing some subtlety?


```suggestion
            $loadContent = ($isActiveTab && !$loadInitialTabWithAjax) || !$obj->supportsAjax();
            $tabItem['content'] = $loadContent ? $this->getView()->record($driver)->getTab($obj) : '';
            $tabItem['paneAttributes'] = [
                'class' => 'record-tab-pane',
                'data-ajax-url' => $ajaxTabUrl,
            ];
            if ($loadContent) {
```

Reply:
Makes sense to me. I implemented it myself and did not commit your suggestion, because something else changed in that part.